src/pointer-analysis/value_set_dereference.cpp

```diff
@@ -76,9 +124,24 @@ exprt value_set_dereferencet::dereference(const exprt &pointer)
 
   std::list<valuet> values;
 
+  exprt compare_against_pointer = pointer;
+
+  if(retained_values.size() >= 2 && should_use_local_definition_for(pointer))
+  {
+    symbolt fresh_binder = get_fresh_aux_symbol(
+      pointer.type(),
+      "derefd_pointer",
```

Comment:
Should use the function_id here

Reply:
I think not, the temporary should not be in the same namespace as the function's locals. Could just be "symex" though, like the namespace occupied by invalid-object symbols